File: src/main/java/com/github/sorusclient/client/adapter/v1_8_9/ScoreboardObjectiveImpl.java
package com.github.sorusclient.client.adapter.v1_8_9;

import com.github.sorusclient.client.adapter.IScoreboardObjective;
import com.github.sorusclient.client.adapter.IScoreboardScore;
import v1_8_9.net.minecraft.scoreboard.ScoreboardObjective;
import v1_8_9.net.minecraft.scoreboard.ScoreboardPlayerScore;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class ScoreboardObjectiveImpl implements IScoreboardObjective {

    private final ScoreboardObjective scoreboardObjective;

    public ScoreboardObjectiveImpl(ScoreboardObjective scoreboardObjective) {
        this.scoreboardObjective = scoreboardObjective;
    }

    @Override
    public List<IScoreboardScore> getScores() {
        List<IScoreboardScore> scores = new ArrayList<>();
        for (ScoreboardPlayerScore score : new ArrayList<>(scoreboardObjective.getScoreboard().getAllPlayerScores(scoreboardObjective))) {
            scores.add(new ScoreboardScoreImpl(score));
        }

        Collections.reverse(scores);

        return scores;
    }

    @Override
    public String getName() {
        return this.scoreboardObjective.getDisplayName();
    }

}

File: src/main/kotlin/com/github/sorusclient/client/adapter/v1_8_9/ScoreboardObjectiveImpl.kt
package com.github.sorusclient.client.adapter.v1_8_9

import com.github.sorusclient.client.adapter.IScoreboardObjective
import com.github.sorusclient.client.adapter.IScoreboardScore
import com.github.sorusclient.client.adapter.IText
import v1_8_9.net.minecraft.scoreboard.ScoreboardObjective
import v1_8_9.net.minecraft.text.BaseText
import v1_8_9.net.minecraft.text.LiteralText
import v1_8_9.net.minecraft.text.Text

class ScoreboardObjectiveImpl(private val scoreboardObjective: ScoreboardObjective) : IScoreboardObjective {
    override val scores: List<IScoreboardScore>
        get() {
            val scores: MutableList<IScoreboardScore> = ArrayList()
            for (score in ArrayList(
                scoreboardObjective.scoreboard.getAllPlayerScores(
                    scoreboardObjective
                )
            )) {
                scores.add(ScoreboardScoreImpl(score))
            }
            scores.reverse()
            return scores
        }
    override val name: IText
        get() = Util.textToApiText(LiteralText(scoreboardObjective.displayName))
}